tools/adventure-pack/src/iteratorPrototypeFilter.ts

```diff
@@ -0,0 +1,26 @@
+import { iteratorPrototype } from "./iteratorPrototype";
+import { iteratorToIterable } from "./iteratorToIterable";
+
+declare global {
+  interface Iterator<T> {
+    filter(
+      callbackfn: (value: T, index: number, array: T[]) => any,
+      thisArg?: any,
```

Comment:
We also don't need to support the `thisArg`, I think this is a vestige of the world before arrow functions, when functions would "forget" which context they belonged to and we needed to remind them.